deno_webgpu/src/lib.rs

```diff
@@ -240,7 +240,10 @@ pub async fn op_webgpu_request_adapter(
         state.put(wgpu_core::hub::Global::new(
             "webgpu",
             wgpu_core::hub::IdentityManagerFactory,
-            backends,
+            wgpu_types::InstanceOptions {
+                backends,
+                dx12_shader_compiler: wgpu_types::Dx12Compiler::Fxc,
```

Comment:
I am not sure what exactly this PR does as I am not familiar with this, however I see that in wgpu its going to be configurable via an env var; would it make sense for this to be configurable for deno, or is it not necessary enough of a feature?

Reply:
It's letting users use a better DX12 shader compiler (newer, faster, maintained, allows for newer feature support), but it comes at the cost of requiring users to ship `dxil.dll` and `dxcompiler.dll`, so I just defaulted to using Fxc (the old unmaintained one) and having users opt into [Dxc](https://github.com/microsoft/DirectXShaderCompiler) (the new one). FXC only support up to ShaderModel 5.1, while DXC supports ShaderModel 6.0 and higher.

An example of why you might want to use DXC is if you wanted to add [FSR2](https://github.com/GPUOpen-Effects/FidelityFX-FSR2) to a game engine, as it requires ShaderModel 6.2 (or it can use 6.6 for 64 wide wavefronts), or if you wanted to use DirectX Raytracing (DXR/RTX/etc).

You can look at the [DXC wiki](https://github.com/microsoft/DirectXShaderCompiler/wiki/Shader-Model-6.0) for what features ShaderModel 6.0+ adds.

I wasn't sure if deno even used DX12 on Windows, or if the deno maintainers considered having potentially backwards incompatible shaders (that work with DXC, but not FXC) a worthwhile tradeoff for getting support for 64bit ints, native 16bit types, DXR support, etc.

Reply:
> but it comes at the cost of requiring users to ship `dxil.dll` and `dxcompiler.dll`

so a user of wgpu has to manually link to these, or is this taken care of?

If this is taken care of, and its only a matter of specifying the right enum value, I would say it seems favourable to support via a `DENO_WEBGPU_DX12_COMPILER` env var

Reply:
The wgpu user has to do it, but linking them just requires dropping the .dlls in the folder, and making sure they're included in what's sent to the end user (that eg. plays the game that uses wgpu and the dxc compiler backend).

Reply:
hm, i guess for now lets leave it as-is